### PR TITLE
Update tox configuration for building docs

### PR DESCRIPTION
**Pull request by namurphy**
This makes/proposes a few changes:

 - I added a new test, `build_docs_nitpicky`, that issues warnings for missing links.  
 - I added the `--keep-going` flag when we have the `-W` flag.  This way the test doc build will keep going after a warning, which will let us fix multiple warnings at the same time instead of having to wait for the doc build run multiple times.
 - I changed all doc builds to use `--parallel=auto`.  This is experimental, and does not ~~make the change in GitHub Actions~~ alter our GitHub Actions setup, so as far as I can tell, the doc builds there will still only be done on one core rather than in parallel.  (edited as a clarification)

### CODE REVIEW

**Comment by namurphy** [comment]
Building the docs in nitpicky mode:
```bash
build finished with problems, 1466 warnings.
```
Huh.

**Review by StanczakDominik — APPROVED** [review]
The changes themselves look good! I can approve it now while we wait for the other PRs to go into main.

File: tox.ini [hunk 1]
```diff
@@ -41,14 +41,21 @@ changedir = {toxinidir}
 extras = docs
 setenv =
     HOME = {envtmpdir}
-commands = sphinx-build docs docs{/}_build{/}html -W -b html
+commands = sphinx-build docs docs{/}_build{/}html -W --keep-going -j=auto -b html
```
Comment (StanczakDominik) [inline tox.ini:44]:
```suggestion
commands = sphinx-build docs docs{/}_build{/}html -W --keep-going -b html {posargs}
```

Comment (StanczakDominik) [reply tox.ini:44]: `tox -- -j=auto` in github action workflow

**Comment by namurphy** [comment]
From today's meeting...
 - [ ] don't set auto-parallelization by default.

**Review by StanczakDominik — APPROVED** [review]

File: docs/development/doc_guide.rst [hunk 2]
```diff
@@ -102,20 +102,38 @@ You can use tox_ to locally build the documentation by running:
...
+
+   tox -e build_docs_nitpicky
+
+.. tip::
+
+   When writing documentation, please make sure to fix any warnings that
+   arise. To enforce this, the ``build_docs`` and ``build_docs_nitpicky``
+   tox_ environments will fail after completing the documentation build
+   if they encountered any warnings (via the ``-W`` and ``--keep-going``
+   flags to `sphinx-build`_).
```
Comment (StanczakDominik) [inline docs/development/doc_guide.rst:136]: Would you like to add a github action for `nitpicky`?

Comment (namurphy) [reply docs/development/doc_guide.rst:136]: Not yet, but I would really like to once all of the ∼10³ `nitpicky` warnings → errors are resolved.  That's going to take some time and might also depend on #1158 (and/or maybe #1149?) to be resolved to define what `optional`, etc. mean in the parameters section of docstrings.  Eventually a GitHub Action for `nitpicky` would be great!

File: changelog/1206.trivial.rst [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Updated tox_ test environments for building the documentation. Added the
```
Comment (rocco8773) [inline changelog/1206.trivial.rst:1]: split into 2 change logs

File: docs/development/doc_guide.rst [hunk 4]
```diff
@@ -102,20 +102,38 @@ You can use tox_ to locally build the documentation by running:
...
+You can shorten the documentation build by running:
 
-You can shorten the tox_ documentation build by running:
+.. code-block:: bash
+
+   tox -e build_docs -- -j=auto
+
+This command will build the documentation in parallel, with the number
+of processes being automatically determined. Flags that follow the
+``--`` are passed to `sphinx-build`_.
```
Comment (rocco8773) [inline docs/development/doc_guide.rst:113]:
```suggestion
Th `-j command 
<https://www.sphinx-doc.org/en/master/man/sphinx-build.html#cmdoption-sphinx-build-j>`_
will build the documentation in parallel, with the number
of processes being automatically determined. Flags that follow the
``--`` are passed to `sphinx-build`_.
```

**Review by rocco8773 — APPROVED** [review]

**Comment by namurphy** [comment]
pre-commit.ci autofix

**Pull request merged**